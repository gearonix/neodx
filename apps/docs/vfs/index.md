# @neodx/vfs

::: danger WIP
`@neodx/vfs` is under redesign, please, check [npm package](https://www.npmjs.com/package/@neodx/vfs) for information about current API.
:::
