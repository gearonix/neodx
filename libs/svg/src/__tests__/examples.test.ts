import { deepReadDir } from '@neodx/fs';
import { getChangesHash } from '@neodx/vfs/testing-utils';
import { describe, expect, test } from 'vitest';
import type { GenerateParams } from '..';
import { generateExample, getExamplesNames } from './testing-utils';

describe('examples', () => {
  const examples = getExamplesNames();
  const optionsMap: Record<string, Partial<GenerateParams>> = {
    'groups-without-root': {
      group: true
    },
    'groups-with-root': {
      group: true,
      root: 'assets'
    },
    react: {
      group: true,
      root: 'assets'
    },
    colors: {
      root: 'assets',
      resetColors: {
        replace: {
          from: ['#000000', '#ffffff', '#000', '#fff', 'black', 'white'],
          to: 'currentColor'
        }
      }
    },
    'experimental-runtime': {
      root: 'assets',
      group: true,
      fileName: '{name}.{hash:8}.svg',
      experimentalRuntime: true
    }
  };

  test.each(examples)(`"%s" example should replay same output`, async name => {
    const { vfs } = await generateExample(name, false, optionsMap[name]);

    expect(await getChangesHash(vfs)).toMatchSnapshot();
  });

  test.each(examples)(`"%s" example should generate files`, async name => {
    const { vfs } = await generateExample(name, true, optionsMap[name]);

    expect(await deepReadDir(vfs.root, { absolute: false })).toMatchSnapshot();
  });
});
