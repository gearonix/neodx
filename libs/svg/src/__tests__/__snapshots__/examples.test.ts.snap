// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`examples > "colors" example should generate files 1`] = `
[
  "assets/custom.svg",
  "assets/fill.svg",
  "assets/mixed.svg",
  "assets/stroke.svg",
  "generated/sprite-info.ts",
  "generated/sprite.svg",
]
`;

exports[`examples > "colors" example should replay same output 1`] = `
[
  [
    "generated/sprite.svg",
    "62b41de3545a4350fe7ef5f3cc3adbbb221c82ef0863bf5871361f821b6cd028",
  ],
  [
    "generated/sprite-info.ts",
    "e90c943544b0deb31f987891a7c4895d28c0bd94b4373696a80ae85074f18c02",
  ],
]
`;

exports[`examples > "colors-advanced" example should generate files 1`] = `
[
  "assets/close.svg",
  "assets/exit.svg",
  "assets/favourite.svg",
  "assets/flags/ac.svg",
  "assets/flags/ad.svg",
  "assets/flags/ae.svg",
  "assets/flags/af.svg",
  "assets/folder-colored.svg",
  "assets/logos/linkedin.svg",
  "assets/logos/twitter.svg",
  "assets/sort-by-visibility.svg",
  "generated/flags.svg",
  "generated/logos.svg",
  "generated/sprite-info.ts",
  "generated/sprite.svg",
]
`;

exports[`examples > "colors-advanced" example should replay same output 1`] = `
[
  [
    "generated/sprite.svg",
    "ffef88ad150b16c58a121b4efe656db2159b9a3498efb53067180a88a0bfd711",
  ],
  [
    "generated/flags.svg",
    "f4756b243c9212d40aa8c5484660ff8dfae36f09c777f89555f557382f453857",
  ],
  [
    "generated/logos.svg",
    "6c76e8cdf522e8b7cb20f1fb6303584d547ff0f15054b4333c351f4f1fb1d2c6",
  ],
  [
    "generated/sprite-info.ts",
    "ba8fc4565b41a227a9cc7372fed0db893a4f52bae852b23bbec89c97ae24623f",
  ],
]
`;

exports[`examples > "groups-with-root" example should generate files 1`] = `
[
  "assets/common/close.svg",
  "assets/common/favourite.svg",
  "assets/format/align-left.svg",
  "assets/format/tag.svg",
  "generated/common.svg",
  "generated/format.svg",
  "generated/sprite-info.ts",
]
`;

exports[`examples > "groups-with-root" example should replay same output 1`] = `
[
  [
    "generated/common.svg",
    "2eb4b56ff266279ca9c0af3b16f7f3114ae35c4555657bb106acbd15ff9d4f52",
  ],
  [
    "generated/format.svg",
    "c890959d268bf53d6006012438788b4104189d3cfaf997b0b43b2fd4f1a41159",
  ],
  [
    "generated/sprite-info.ts",
    "86b30a6c62260f488f7840660677e396f97aeefd2e1377ab0f7603e128257459",
  ],
]
`;

exports[`examples > "groups-without-root" example should generate files 1`] = `
[
  "assets/common/close.svg",
  "assets/common/favourite.svg",
  "assets/format/align-left.svg",
  "assets/format/tag.svg",
  "generated/assets/common.svg",
  "generated/assets/format.svg",
  "generated/sprite-info.ts",
]
`;

exports[`examples > "groups-without-root" example should replay same output 1`] = `
[
  [
    "generated/assets/common.svg",
    "2eb4b56ff266279ca9c0af3b16f7f3114ae35c4555657bb106acbd15ff9d4f52",
  ],
  [
    "generated/assets/format.svg",
    "c890959d268bf53d6006012438788b4104189d3cfaf997b0b43b2fd4f1a41159",
  ],
  [
    "generated/sprite-info.ts",
    "ff73fe4f93a997f28451d9096746a470c68939807d28585f137f89397635e1cd",
  ],
]
`;

exports[`examples > "metadata" example should generate files 1`] = `
[
  "assets/common/close.svg",
  "assets/common/favourite.svg",
  "assets/format/align-left.svg",
  "assets/format/tag.svg",
  "generated/common.2eb4b56f.svg",
  "generated/format.c890959d.svg",
  "generated/meta.ts",
]
`;

exports[`examples > "metadata" example should replay same output 1`] = `
[
  [
    "generated/common.2eb4b56f.svg",
    "2eb4b56ff266279ca9c0af3b16f7f3114ae35c4555657bb106acbd15ff9d4f52",
  ],
  [
    "generated/format.c890959d.svg",
    "c890959d268bf53d6006012438788b4104189d3cfaf997b0b43b2fd4f1a41159",
  ],
  [
    "generated/meta.ts",
    "c7f023721dbcde44c77868aa1204555fdef83176f944624c55114967fcf815e8",
  ],
]
`;

exports[`examples > "react" example should generate files 1`] = `
[
  "assets/common/close.svg",
  "assets/common/favourite.svg",
  "assets/format/align-left.svg",
  "assets/format/tag.svg",
  "example.tsx",
  "generated/common.svg",
  "generated/format.svg",
  "generated/sprite-info.ts",
  "icon.tsx",
]
`;

exports[`examples > "react" example should replay same output 1`] = `
[
  [
    "generated/common.svg",
    "2eb4b56ff266279ca9c0af3b16f7f3114ae35c4555657bb106acbd15ff9d4f52",
  ],
  [
    "generated/format.svg",
    "c890959d268bf53d6006012438788b4104189d3cfaf997b0b43b2fd4f1a41159",
  ],
  [
    "generated/sprite-info.ts",
    "86b30a6c62260f488f7840660677e396f97aeefd2e1377ab0f7603e128257459",
  ],
]
`;

exports[`examples > "simple" example should generate files 1`] = `
[
  "assets/arrow-drop-down.svg",
  "assets/arrow-drop-up.svg",
  "generated/sprite-info.ts",
  "generated/sprite.svg",
]
`;

exports[`examples > "simple" example should replay same output 1`] = `
[
  [
    "generated/sprite.svg",
    "de338dcd1643d85f1cc9bc1b0fd01e6904bb82b600a1a4914c31577766ead4d2",
  ],
  [
    "generated/sprite-info.ts",
    "1794ca9969cc76ca040786822e6d31b0070af59bc05062d9855ce3e12aae2c06",
  ],
]
`;
